### ISSUE TITLE
[steamshovel] ipdf I3DOMLikelihoodArtist causes shovelart to segfault on exit (Trac #1711)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1703]
Simply loading `icecube.ipdf.artists` causes `shovelart.so` to segfault on exit. It is not clear why this is happening. This code was disabled in r146010/IceCube, but the issue still needs to be fixed. 

This segfault was causing the doc build to segfault and fail even though the docs build correctly. See the complete stacktrace at http://builds.icecube.wisc.edu/builders/docs/builds/59/steps/compile_2/logs/stdio

Migrated from https://code.icecube.wisc.edu/ticket/1711
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:55", 
    "description": "Simply loading `icecube.ipdf.artists` causes `shovelart.so` to segfault on exit. It is not clear why this is happening. This code was disabled in r146010/IceCube, but the issue still needs to be fixed. \n\nThis segfault was causing the doc build to segfault and fail even though the docs build correctly. See the complete stacktrace at http://builds.icecube.wisc.edu/builders/docs/builds/59/steps/compile_2/logs/stdio", 
    "reporter": "kjmeagher", 
    "cc": "nega", 
    "resolution": "fixed", 
    "_ts": "1550067295757382", 
    "component": "combo reconstruction", 
    "summary": "[steamshovel] ipdf I3DOMLikelihoodArtist causes shovelart to segfault on exit", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-05-18T09:12:15", 
    "milestone": "", 
    "owner": "hdembinski", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160518T15:13:34: hdembinski changed component from "cmake" to "icerec"

**Comment by icecube-trac** [comment]
Trac update at 20160518T15:19:13: hdembinski commented:

> Cannot be confirmed on Mac El Capitan. This is what I tried:
> 
> - open empty steamshovel
> - ctrl+T to open shell
> - import icecube.ipdf.artists
> - adding an I3DOMLikelihoodArtist to the scene
> - close steamshovel
> 
> I tried this several times, no segfault.
> 
> Please provide specific instructions on how to reproduce the bug.

**Comment by icecube-trac** [comment]
Trac update at 20160518T15:20:01: hdembinski commented:

> Nevermind, on the third try i got the segfault.

**Comment by icecube-trac** [comment]
Trac update at 20160518T16:20:44: hdembinski commented:

> Or maybe not. I ran this script like a few dozen times:
> 
> from time import sleep
> import icecube.ipdf.artists
> 
> scenario = window.gl.scenario
> artist = scenario.add( "I3DOMLikelihoodArtist", [] )
> sleep( 1 )
> 
> app.quit( 0 )
> 
> I don't get the segfault.

**Comment by icecube-trac** [comment]
Trac update at 20160518T17:11:09: kjmeagher commented:

> to retproduce just run `python -c 'import icecube.ipdf.artists'`

**Comment by icecube-trac** [comment]
Trac update at 20160518T19:55:31: 
* hdembinski changed resolution from "" to "fixed"
* hdembinski changed status from "new" to "closed"
* hdembinski commented:

> In changeset 146075:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="146075"
> fixes segfault at exit, closes #1703
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:55: 
* olivas changed milestone from "IceRec 5.2" to ""
* olivas commented:

> Milestone IceRec 5.2 deleted
